README.md

```diff
@@ -15,6 +15,15 @@ depend on the number of vertices, the number of edges, or both?
 Describe your reasoning and the conclusion you've come to. Your reasoning is the
 most important part. Add your answer to this markdown file.
 
+The runtime complexity depends on both the number of vertices (V) and the number of edges (E).
+
+1. initializing an adjacency list, with each vertex have an empty spot, one loop involved, so the time complexity is $O(V)$
+2. filling the adjacency list needs two nested loops, the outer loop iterates each vertex in the row, the inner loop iterates each vertex in the column, combine both loops, which resulting a time complexity $O(V^2)$. Moreover, the complexity of checking the directed edge is $O(1)$, add the vertex to the list is also $O(1)$.
+3. finalizing the list used one loop, which runs V times, and inside the loop, the "Set" remove the dupulicates, results $O(E)$, and the "Sort" ensures the constant order, which result the time complexity to be $O(E log V)$ for sorting each list
```

Comment:
Not following you here -- why is there a V in the complexity for sorting the list of edges?